internal/subscription/core.go

```diff
@@ -68,12 +68,45 @@ func (c *Core) CreateSubscription(ctx context.Context, m *Model) error {
 		logger.Ctx(ctx).Errorw("topic project not found", "name", m.ExtractedTopicProjectID)
 		return merror.New(merror.NotFound, "project not found")
 	}
-	if ok, err = c.topicCore.ExistsWithName(ctx, m.Topic); !ok {
+
+	var topicModel *topic.Model
+	if topicModel, err = c.topicCore.Get(ctx, m.GetTopic()); err != nil {
 		if err != nil {
 			return err
 		}
-		return merror.Newf(merror.NotFound, "topic with name %s not found", m.Topic)
 	}
+
+	// for subscription over deadletter topics, skip the retry and deadletter topic creation
+	if topicModel.IsDeadLetterTopic() == false {
+		// create retry topic for subscription
+		// TODO: update based on retry policy
+		err = c.topicCore.CreateRetryTopic(ctx, &topic.Model{
+			Name:               m.GetRetryTopic(),
+			ExtractedTopicName: m.ExtractedSubscriptionName + topic.RetryTopicSuffix,
+			ExtractedProjectID: m.ExtractedTopicProjectID,
+			NumPartitions:      topicModel.NumPartitions,
+		})
+
+		if err != nil {
+			logger.Ctx(ctx).Errorw("failed to create retry topic for subscription", "name", m.GetRetryTopic())
+			return err
+		}
+
+		// create deadletter topic for subscription
+		// TODO: read the deadletter policy and update occordingly
+		err = c.topicCore.CreateDeadLetterTopic(ctx, &topic.Model{
+			Name:               m.GetDeadLetterTopic(),
+			ExtractedTopicName: m.ExtractedSubscriptionName + topic.DeadLetterTopicSuffix,
+			ExtractedProjectID: m.ExtractedTopicProjectID,
+			NumPartitions:      topicModel.NumPartitions,
+		})
+
+		if err != nil {
+			logger.Ctx(ctx).Errorw("failed to create deadletter topic for subscription", "name", m.GetDeadLetterTopic())
```

Comment:
```suggestion
			logger.Ctx(ctx).Errorw("failed to create deadletter topic for subscription", "name", m.GetDeadLetterTopic(), "error" , err.Error())
```
log the actual error as well. Same above.